### PR TITLE
Game Over Bug

### PR DESCRIPTION
**Pull request by kimberlypn**
Changes:
- Fixed a bug where if a user left after the game over page, then the other user's page would switch to the waiting page. This happened because leaving reset the state to 0 players, and the waiting page appears when there are less than 2 players. Fixed this by only removing the player when s/he leaves and resetting the entire game state once both players have left
- If a user tries to join a game that is over but has not yet been reset, s/he will see a "Game is currently busy. Try again later" message and a link to go back to the game name page.
- Fixed a rounding issue on the leaderboard. I was only rounding the ratio to 2 decimal places, but I had to round the ratio * 100 to 2 decimal places; otherwise, I could still end up with a bunch of decimals.
- Removed .gif files. Battle PNGs seem to be proportionally correct now.
- Right-aligned bottom player's stuff to make things look more symmetric

Bugs:
- In-game HELP link does not work anymore. Need to figure out where to place the "battlefield" id

### CODE REVIEW

**Review by mdang8 — COMMENTED** [review]
When I play and a player reaches 0 health, the game doesn't end. The game continues and the losing player's health will go into negative numbers with each loss after that.

File: shambomon/assets/js/shambomon.jsx [hunk 1]
```diff
@@ -96,37 +96,34 @@ class Shambomon extends React.Component {
...
     let ready = this.isReady();
-    let winner = this.hasWinner();
     // Game has less than two players
-    if (!ready) {
+    if (!ready && !this.state.gameOver) {
       return <Waiting />;
     }
     // Someone has won
-    else if (winner) {
+    else if (this.state.gameOver) {
```
Comment (kimberlypn) [inline shambomon/assets/js/shambomon.jsx:121]: `this.state.gameOver` gets set to `true` on the server if either player's HP is <= 0. Before, this check was done on the browser.

File: shambomon/lib/shambomon/game.ex [hunk 2]
```diff
@@ -141,6 +144,11 @@ defmodule Shambomon.Game do
       game = update_messages(game, Map.get(p2, :char), health_decr)
     end
 
+    # Set the gameOver flag if applicable
+    if (Map.get(p1, :health) <= 0) or Map.get(p2, :health <= 0) do
```
Comment (kimberlypn) [inline shambomon/lib/shambomon/game.ex]: This is where the `gameOver` flag is set.

**Review by kimberlypn — COMMENTED** [review]
Hm... I tried again on my computer, and it works fine. I highlighted the spots in the code that should prevent that from happening.

File: shambomon/lib/shambomon/game.ex (hunk 2 above)
Comment (kimberlypn) [reply shambomon/lib/shambomon/game.ex]: Parenthesis was in the wrong spot on that second condition, so the bug occurs when p2 dies. rip

**Pull request merged**